src/services/preProcess.ts

```diff
@@ -345,6 +345,23 @@ namespace ts {
                     break;
                 }
 
+                if (scanner.getToken() === SyntaxKind.TemplateHead) {
+                    while (true) {
+                        if (scanner.getToken() === SyntaxKind.EndOfFileToken) {
+                            break;
+                        }
+                        if (scanner.getToken() === SyntaxKind.ImportKeyword) {
+                            tryConsumeImport();
+                        }
+                        if (scanner.getToken() === SyntaxKind.CloseBraceToken) {
+                            scanner.reScanTemplateToken(/*isTaggedTemplate*/ false);
+                            break;
+                        }
+                        scanner.scan();
+                    }

```

Comment:
It'd be good to handle arbitrary nesting of template string interpolations too.

```suggestion
                    let templateDepth = 1;
                    while (templateDepth) {
                        if (scanner.getToken() === SyntaxKind.EndOfFileToken) {
                            break;
                        }
                        else if (scanner.getToken() === SyntaxKind.ImportKeyword) {
                            tryConsumeImport();
                        }
                        else if (scanner.getToken() === SyntaxKind.TemplateHead) {
                            templateDepth++;
                        }
                        else if (scanner.getToken() === SyntaxKind.CloseBraceToken) {
                            const templateToken = scanner.reScanTemplateToken(/*isTaggedTemplate*/ false);
                            if (templateToken === SyntaxKind.TemplateTail) {
                                templateDepth--;
                            }
                        }
                        scanner.scan();
                    }
```

Also consider switching to a `switch`/`case` on `scanner.getToken()`.

Reply:
I’ve added this case to tests